configs/components/puppet-ca-bundle.rb

```diff
@@ -18,9 +18,7 @@
 
   # Include the keystore for the agent if the agent has a java
   if settings[:runtime_project] == 'agent'
-    if platform.name =~ /el-(6|7|8)|sles-(11|12)|debian-(9|10)|ubuntu-(14|15|16|18)/
-      install_commands << "#{platform[:make]} keystore OPENSSL=#{openssl_cmd} DESTDIR=#{File.join(settings[:prefix], 'ssl')}"
-    end
+    install_commands << "if [[ `which keytool` ]]; then #{platform[:make]} keystore OPENSSL=#{openssl_cmd} DESTDIR=#{File.join(settings[:prefix], 'ssl')}; fi"
```

Comment:
Ah yeah makes sense. Will this work for cross compiled platforms? Will Windows do something silly if Java happens to be installed?

Reply:
wrt cross compiled I don't think is an issue because the default format for many platforms (and what the server expects to read the keystore as) is JKS, which I believe is a cross platform Java standard.

However, it looks like starting in Java 9 the default format for creating a keystore was changes PKCS12, which won't work for Puppet Server. I've created this PR to correct that: https://github.com/puppetlabs/puppet-ca-bundle/pull/14

Also, the JKS format isn't FIPS compatible and a FIPS PE build will refuse to load this file. I know if you tell Java to load a JKS file but actually give it a PKCS12 file it simply say it found no certs (but not explode), I need to verify that happens on FIPS as well. To generate the keystore type we use in FIPS (BCFKS) we need the BouncyCastleFIPS library available. I think skipping FIPS (or shipping a file it won't read) as part of this is fine as talking to arbitrary hosts on the internet isn't usually what FIPS users want anyways (quite the opposite). FIPS users may regenerate this themselves using the BCFIPS jars we provide, and without extra work FIPS users should still be able to authenticate to internal hosts using the `ssl_trust_store` option.

Reply:
Confirmed that windows-2012r2-x64 and windows-2019-x64 puppet-runtime-main builds. `puppet-agent` builds with windows-2012r2-x64 then installs correctly but does not have a keystore in the location I would expect it `C:\Program Files\Puppet Labs\Puppet\puppet\ssl`
![Screenshot from 2022-05-12 14-42-03](https://user-images.githubusercontent.com/241423/168178263-b499f287-2398-44d1-ab10-b3bfbbce377a.png)

Reply:
> I can add back a conditional to skip this on FIPS builds?

Yeah explicitly omitting the file sounds better than creating the file in a format that BouncyCastleFIPS can't load. So even if a user regenerates the file in the correct format, it'd be overwritten when the puppet-agent package is updated.

Reply:
updated with a fips exclusion and a healthy comment as to why (updated commit message too).
built and confirmed no puppet-cacerts file on redhatfips-7-x86_64 (aptly named `epic-correction.delivery.puppetlabs.net`).